Httpx Read Errors Occur in K8s Watcher

These seem to happen periodically but often in the k8s watcher,

```
2026-01-16T06:10:14.013633 [ERROR] 7/MainThread renku_data_services.k8s.watcher.core (core.py:122) [-] - watch loop failed for GVK(kind='AmaltheaSession',
 version='v1alpha1', group='amalthea.dev') in cluster 0RENK1RENK2RENK3RENK4RENK5
Traceback (most recent call last):
  File "/app/env/lib/python3.13/site-packages/httpx/_transports/default.py", line 101, in map_httpcore_exceptions
    yield
  File "/app/env/lib/python3.13/site-packages/httpx/_transports/default.py", line 271, in __aiter__
    async for part in self._httpcore_stream:
        yield part
  File "/app/env/lib/python3.13/site-packages/httpcore/_async/connection_pool.py", line 407, in __aiter__
    raise exc from None
  File "/app/env/lib/python3.13/site-packages/httpcore/_async/connection_pool.py", line 403, in __aiter__
    async for part in self._stream:
        yield part
  File "/app/env/lib/python3.13/site-packages/httpcore/_async/http11.py", line 342, in __aiter__
    raise exc
  File "/app/env/lib/python3.13/site-packages/httpcore/_async/http11.py", line 334, in __aiter__
    async for chunk in self._connection._receive_response_body(**kwargs):
        yield chunk
  File "/app/env/lib/python3.13/site-packages/httpcore/_async/http11.py", line 203, in _receive_response_body
    event = await self._receive_event(timeout=timeout)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/env/lib/python3.13/site-packages/httpcore/_async/http11.py", line 217, in _receive_event
    data = await self._network_stream.read(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        self.READ_NUM_BYTES, timeout=timeout
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/app/env/lib/python3.13/site-packages/httpcore/_backends/anyio.py", line 32, in read
    with map_exceptions(exc_map):
         ~~~~~~~~~~~~~~^^^^^^^^^
  File "/usr/local/lib/python3.13/contextlib.py", line 162, in __exit__
    self.gen.throw(value)
    ~~~~~~~~~~~~~~^^^^^^^
  File "/app/env/lib/python3.13/site-packages/httpcore/_exceptions.py", line 14, in map_exceptions
    raise to_exc(exc) from exc
httpcore.ReadError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/env/lib/python3.13/site-packages/renku_data_services/k8s/watcher/core.py", line 112, in __watch_kind
    async for event_type, obj in watch:
    ...<5 lines>...
            await self.__handler(cluster.with_api_object(obj), event_type)
  File "/app/env/lib/python3.13/site-packages/kr8s/_api.py", line 587, in async_watch
    async for line in response.aiter_lines():
    ...<12 lines>...
        yield event["type"], obj
  File "/app/env/lib/python3.13/site-packages/httpx/_models.py", line 1031, in aiter_lines
    async for text in self.aiter_text():
        for line in decoder.decode(text):
            yield line
  File "/app/env/lib/python3.13/site-packages/httpx/_models.py", line 1018, in aiter_text
    async for byte_content in self.aiter_bytes():
    ...<2 lines>...
            yield chunk
  File "/app/env/lib/python3.13/site-packages/httpx/_models.py", line 997, in aiter_bytes
    async for raw_bytes in self.aiter_raw():
    ...<2 lines>...
            yield chunk
  File "/app/env/lib/python3.13/site-packages/httpx/_models.py", line 1055, in aiter_raw
    async for raw_stream_bytes in self.stream:
    ...<2 lines>...
            yield chunk
  File "/app/env/lib/python3.13/site-packages/httpx/_client.py", line 176, in __aiter__
    async for chunk in self._stream:
        yield chunk
  File "/app/env/lib/python3.13/site-packages/httpx/_transports/default.py", line 270, in __aiter__
    with map_httpcore_exceptions():
         ~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/local/lib/python3.13/contextlib.py", line 162, in __exit__
    self.gen.throw(value)
    ~~~~~~~~~~~~~~^^^^^^^
  File "/app/env/lib/python3.13/site-packages/httpx/_transports/default.py", line 118, in map_httpcore_exceptions
    raise mapped_exc(message) from exc
httpx.ReadError

```

--- a/components/renku_data_services/k8s/watcher/core.py
+++ b/components/renku_data_services/k8s/watcher/core.py
@@ -121,8 +121,6 @@ class K8sWatcher:
                         await self.__handler(cluster.with_api_object(obj), event_type)
             except ValueError:
                 pass
-            except Exception as e:
-                logger.error(f"watch loop failed for {kind} in cluster {cluster_id}", exc_info=e)
             except (httpx.ReadError, httpcore.ReadError):
                 # This can happen occasionally - most likely means that the k8s cluster stopped the connection
                 logger.warning(
@@ -130,6 +128,8 @@ class K8sWatcher:
                     f"watch for cluster {cluster_id} and kind {kind}."
                 )
                 continue
+            except Exception as e:
+                logger.error(f"watch loop failed for {kind} in cluster {cluster_id}", exc_info=e)
 
             # Add a sleep to prevent retrying in a loop the same action instantly.
             await asyncio.sleep(10)

